lib/backend/src/Obelisk/Backend.hs

```diff
@@ -263,29 +312,111 @@ renderGhcjsFrontend urlEnc ghcjsWidgets route configs f = do
 
 -- | Preload all.js in a link tag.
 -- This is the default preload method.
-preloadGhcjs :: Text -> FrontendWidgetT r ()
-preloadGhcjs allJsUrl = elAttr "link" ("rel" =: "preload" <> "as" =: "script" <> "href" =: allJsUrl) blank
+preloadGhcjs :: GhcjsWasmAssets -> FrontendWidgetT r ()
+preloadGhcjs (GhcjsWasmAssets allJsUrl mWasm) = case mWasm of
+  Nothing -> elAttr "link" ("rel" =: "preload" <> "as" =: "script" <> "href" =: allJsUrl) blank
+  Just wasmAssets -> scriptTag $ view _1 (wasmScripts allJsUrl wasmAssets)
 
 -- | Load the script from the given URL in a deferred script tag.
 -- This is the default method.
-deferredGhcjsScript :: Text -> FrontendWidgetT r ()
-deferredGhcjsScript allJsUrl = elAttr "script" ("type" =: "text/javascript" <> "src" =: allJsUrl <> "defer" =: "defer") blank
+deferredGhcjsScript :: GhcjsWasmAssets -> FrontendWidgetT r ()
+deferredGhcjsScript (GhcjsWasmAssets allJsUrl mWasm) = case mWasm of
+  Nothing -> elAttr "script" ("type" =: "text/javascript" <> "src" =: allJsUrl <> "defer" =: "defer") blank
+  Just wasmAssets -> scriptTag $ view _2 (wasmScripts allJsUrl wasmAssets)
+
+scriptTag :: DomBuilder t m => Text -> m ()
+scriptTag t = elAttr "script" ("type" =: "text/javascript") $ text t
+
+wasmScripts
+  :: Text
+  -- ^ URL (could be relative) of "all.js"
+  -> (Text, WasmAssets)
+  -- ^ Root URL (could be relative) of wasm assets along with
+  -- the record of wasm assets.
+  -> (Text, Text, (Int -> Text))
+  -- ^ (preload script, deferred run script, delayed run script (Int input is the delay))
+wasmScripts allJsUrl' (wasmRoot, wAssets) = (preloadScript, runJsScript, delayedJsScript)
+  where
+    wrapName f = "'" <> wasmRoot <> "/" <> f wAssets <> "'"
+    jsaddleJs = wrapName _wasmAssets_jsaddleJs
+    interfaceJs = wrapName _wasmAssets_interfaceJs
+    runnerJs = wrapName _wasmAssets_runnerJs
+    wasmUrl = wrapName _wasmAssets_wasmFile
+    allJsUrl = "'" <> allJsUrl' <> "'"
+
+    preloadScript =
+      "add_preload_tag = function (docSrc) {\
+        \var link_tag = document.createElement('link');\
+        \link_tag.rel = 'preload';\
+        \link_tag.as = 'script';\
+        \link_tag.href = docSrc;\
+        \document.head.appendChild(link_tag);\
+      \};\
+      \if (typeof(WebAssembly) === 'undefined') {\
+        \add_preload_tag(" <> allJsUrl <> ");\
+      \} else {\
+        \add_preload_tag(" <> wasmUrl <> ");\
+        \add_preload_tag(" <> jsaddleJs <> ");\
+        \add_preload_tag(" <> interfaceJs <> ");\
+        \add_preload_tag(" <> runnerJs <> ");\
+      \}"
+
+    -- The 'wasmFile' variable is needed here because export API is not working
+    -- in the webabi ts code. The wasm file is fetched by the runnerJs.
+    runJsScript =
+      "add_deferload_tag = function (docSrc) {\
+        \var tag = document.createElement('script');\
+        \tag.type = 'text/javascript';\
+        \tag.src = docSrc;\
+        \tag.setAttribute = ('defer', 'defer');\
```

Comment:
I fear for backwards compatibility on this syntax. Let's stick with the function call. 
```suggestion
        \tag.setAttribute('defer', 'defer');\
```